src/library/styles/__tests__/createStyledComponent.spec.js

```diff
@@ -0,0 +1,121 @@
+/* @flow */
+import React from 'react';
+import { mount } from 'enzyme';
+import { createStyledComponent } from '../../styles';
+
+const mountButton = (props = {}, styles = {}, options = {}) => {
+  const Button = createStyledComponent('button', styles, options);
+  return mount(<Button {...props} />);
+};
+
+describe('createStyledComponent', () => {
+  const wrapper = mountButton({ type: 'button', xyz: true });
+  const button = wrapper.childAt(0);
+
+  it('forwards valid DOM attributes', () => {
+    expect(button.props().type).toBeDefined();
+  });
+
+  it('filters invalid DOM attributes', () => {
+    expect(button.props().xyz).toBeUndefined();
+  });
+
+  describe('with withProps option', () => {
+    const wrapper = mountButton(
+      {},
+      {},
+      {
+        withProps: {
+          type: 'button'
+        }
+      }
+    );
+    const button = wrapper.childAt(0);
+
+    it('passes props to component', () => {
+      expect(button.props().type).toBe('button');
+    });
+  });
+
+  describe('with filterProps option', () => {
+    const wrapper = mountButton(
+      { filterMe: true },
+      {},
+      {
+        filterProps: ['filterMe']
+      }
+    );
+    const button = wrapper.childAt(0);
+
+    it('filters filterProps', () => {
+      expect(button.props().filterMe).toBeUndefined();
```

Comment:
Is this test & expectation testing something different than the one on line 16?

Or, put another way, is testing based on `.props()` sufficient to ensure something is actually in the DOM?